### PR TITLE
feat: upgrade azcopy to 10.21.2

### PR DESCRIPTION
**Pull request by umagnus**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

cleanup: update azcopy 10.21.2

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: pkg/azurefileplugin/Dockerfile [hunk 1]
```diff
@@ -18,17 +18,17 @@ ARG ARCH=amd64
...
 RUN if [ "$ARCH" == "arm64" ] ; then \
-  azcopyURL=${azcopyURL}"-arm64"; fi
-RUN curl -sL ${azcopyURL} | tar -xz && \
+  azcopyURL=https://azcopyvnext.azureedge.net/releases/release-10.21.2-20231106/azcopy_linux_arm64_10.21.2.tar.gz; fi
+RUN wget -O azcopy.tar.gz ${azcopyURL} && \
+  tar xvzf azcopy.tar.gz -C . && rm azcopy.tar.gz && \
   mv ./azcopy_linux_$ARCH_*/azcopy /usr/local/bin/azcopy && \
   rm -rf ./azcopy_linux_$ARCH_*
 RUN chmod +x /usr/local/bin/azcopy
-RUN apt remove curl -y
```
Comment (andyzhangx) [inline pkg/azurefileplugin/Dockerfile:31]: if you are using wget, you need to remove wget finally

Comment (umagnus) [reply pkg/azurefileplugin/Dockerfile:31]: done

**Review by andyzhangx — COMMENTED** [review]
/ok-to-test

**Review by andyzhangx — APPROVED** [review]

**Pull request merged**